natural_language_processing/word_embeddings/word_embeddings.py

```diff
@@ -33,8 +37,7 @@ def analogy(self, x1: str, x2: str, y1: str) -> str:
         x1, x2, y1 = x1.lower(), x2.lower(), y1.lower()
         error_msg = 'every word must be in the vocabulary'
         assert all(self.is_in_vocab(x) for x in (x1, x2, y1)), error_msg
-        result = self._wv.most_similar(positive=[y1, x2], negative=[x1])[0][0]
-        return result
+        return self._wv.most_similar(positive=[y1, x2], negative=[x1])[0][0]
 
     def n_similarity(self, list1: list, list2: list) -> float:
```

Comment:
Add proper type hints in the functions (see [here](https://docs.python.org/3/library/typing.html))

Reply:
__Please__ Add proper type hints in the functions...